test/blocking-links/shadow-dom.html

```diff
@@ -0,0 +1,111 @@
+    <link rel="import" href="../../starcounter-include.html">
+    <script src="../helpers/WCT-Polyfill_bugs_workaround.js"></script>
+    <link rel="import" href="../../../enlighted-link/enlighted-link.html">
+</head>
+
+<body>
+
+    <div id="testFixtureConatiner"></div>
+
+    <script>
+        const COMPOSITIONS = {
+            BLOCKING:{
+                "stylesheet": `<link rel="stylesheet" href="/components/starcounter-include/sc/blocking-links/styles.css">`,
+                "enlighted-link": `<enlighted-link rel="import" href="/components/starcounter-include/sc/blocking-links/import.html"></enlighted-link>`
+            },
+            ASYNC:{
+                "stylesheet": `<link async rel="stylesheet" href="/components/starcounter-include/sc/blocking-links/styles.css">`,
+                "enlighted-link": `<enlighted-link async rel="import" href="/components/starcounter-include/sc/blocking-links/import.html"></enlighted-link>`
+            }
+        };
+
+        var scInclude, importedTemplate;
+        describe(`imported content`,()=>{
+            beforeEach(function (done) {
+                scInclude = document.createElement('starcounter-include');
+                testFixtureConatiner.appendChild(scInclude);
+                importedTemplate = scInclude.querySelector_template_for_buggy_polyfill();
+                importedTemplate.addEventListener('stamping', ()=>{
+                    setTimeout(done);
+                });
+                scInclude.viewModel = {
+                    App: {
+                        Html: `templateTo.html`,
+                        doesItWork: "worked!"
+                    }
+                };
+            });
+            afterEach(()=>{
+                testFixtureConatiner.innerHTML = '';
+            });
+
+            ['stylesheet'/*, 'enlighted-link'*/].forEach((linkType)=>{
+                describe('when presentation is set (via `updateComposition`) to the one that contains a ', function () {
+                    describe(` \`${linkType}\` element without \`async\` attribute`, function () {
+                        beforeEach(function () {
+                            scInclude.updateComposition(COMPOSITIONS.BLOCKING[linkType]);
+                        });
+
+                        it('should set `loading-presentation` attribute on itself', function () {
+                            expect(scInclude).to.have.HTMLAttribute('loading-presentation').equal("");
+                        });
+                        it('should have `visibility` style set to `hidden`', function () {
+                            expect(scInclude.style).to.have.property('visibility').equal('hidden');
+                        });
+                        describe('after content is stamped, and all imports are loaded', ()=>{
+                            beforeEach(function (done) {
+                                setTimeout(done, 500);
+                            });
```

Comment:
Could you explain we why need to wait this long. I know now, but will probably forget soon.

Reply:
Is there something I already forgot except "and all imports are loaded" ?

Reply:
No I think this would be enough

Reply:
Then this is stated in `describe`